pkg/kubectl/cmd/edit.go

```diff
@@ -162,13 +154,24 @@ func runEdit(f cmdutil.Factory, out, errOut io.Writer, cmd *cobra.Command, args
 
 		containsError := false
 		for {
-			originalObj, err := resource.AsVersionedObject(infos, false, defaultVersion, encoder)
-			if err != nil {
-				return err
+			var originalObj runtime.Object
+			switch len(infos) {
+			case 1:
+				originalObj = infos[0].Object
+			default:
+				l := &unstructured.UnstructuredList{
```

Comment:
Is this common logic that should be associated with some type?

e.g. `type ResourceList []*resource.Info` or something?